sonar-python-plugin/src/main/java/org/sonar/plugins/python/coverage/PythonCoverageSensor.java

```diff
@@ -41,30 +44,46 @@
   private static final Logger LOG = LoggerFactory.getLogger(PythonCoverageSensor.class);
 
   public static final String REPORT_PATH_KEY = "sonar.python.coverage.reportPath";
+  public static final String DEFAULT_REPORT_PATH = "coverage-reports/*coverage-*.xml";
+  // Deprecated report path keys
   public static final String IT_REPORT_PATH_KEY = "sonar.python.coverage.itReportPath";
+  public static final String IT_DEFAULT_REPORT_PATH = "";
   public static final String OVERALL_REPORT_PATH_KEY = "sonar.python.coverage.overallReportPath";
-  public static final String DEFAULT_REPORT_PATH = "coverage-reports/coverage-*.xml";
-  public static final String IT_DEFAULT_REPORT_PATH = "coverage-reports/it-coverage-*.xml";
-  public static final String OVERALL_DEFAULT_REPORT_PATH = "coverage-reports/overall-coverage-*.xml";
+  public static final String OVERALL_DEFAULT_REPORT_PATH = "";
 
   public void execute(SensorContext context) {
     String baseDir = context.fileSystem().baseDir().getPath();
-    Settings settings = context.settings();
+    Configuration config = context.config();
+
+    logDeprecatedPropertyUsage(config, IT_REPORT_PATH_KEY, REPORT_PATH_KEY);
+    logDeprecatedPropertyUsage(config, OVERALL_REPORT_PATH_KEY, REPORT_PATH_KEY);
 
     HashSet<InputFile> filesCovered = new HashSet<>();
     List<File> reports = new ArrayList<>();
-    reports.addAll(getReports(settings, baseDir, REPORT_PATH_KEY, DEFAULT_REPORT_PATH));
-    reports.addAll(getReports(settings, baseDir, IT_REPORT_PATH_KEY, IT_DEFAULT_REPORT_PATH));
-    reports.addAll(getReports(settings, baseDir, OVERALL_REPORT_PATH_KEY, OVERALL_DEFAULT_REPORT_PATH));
+    reports.addAll(getReports(config, baseDir, REPORT_PATH_KEY, DEFAULT_REPORT_PATH));
+    reports.addAll(getReports(config, baseDir, IT_REPORT_PATH_KEY, IT_DEFAULT_REPORT_PATH));
+    reports.addAll(getReports(config, baseDir, OVERALL_REPORT_PATH_KEY, OVERALL_DEFAULT_REPORT_PATH));
     if (!reports.isEmpty()) {
       LOG.info("Python test coverage");
-      for (File report : reports) {
+      for (File report : uniqueAbsolutePaths(reports)) {
         Map<InputFile, NewCoverage> coverageMeasures = parseReport(report, context);
         saveMeasures(coverageMeasures, filesCovered);
       }
     }
   }
 
+  private void logDeprecatedPropertyUsage(Configuration config, String deprecatedKey, String replacementKey) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Make "logDeprecatedPropertyUsage" a "static" method. ([squid:S2325](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.python%3Apython&branch=SONARPY-255&issues=AWEdgoEUZBbiMZMLwWhh&open=AWEdgoEUZBbiMZMLwWhh)

[ISSUE_KEY]: AWEdgoEUZBbiMZMLwWhh